src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.AliasesAndUsings.cs

```diff
@@ -85,14 +84,41 @@ private SingleNamespaceDeclaration GetMatchingNamespaceDeclaration(CSharpSyntaxN
             throw ExceptionUtilities.Unreachable();
         }
 
-        private static AliasesAndUsings GetOrCreateAliasAndUsings(
+        private AliasesAndUsings GetOrCreateAliasAndUsings(
             ref ImmutableSegmentedDictionary<SingleNamespaceDeclaration, AliasesAndUsings> dictionary,
             SingleNamespaceDeclaration declaration)
         {
-            return RoslynImmutableInterlocked.GetOrAdd(
-                ref dictionary,
-                declaration,
-                static _ => new AliasesAndUsings());
+            if (dictionary.IsDefault)
+            {
+                var builder = ImmutableSegmentedDictionary.CreateBuilder<SingleNamespaceDeclaration, AliasesAndUsings>(ReferenceEqualityComparer.Instance);
+#if DEBUG
+                var builderForAsserts = ImmutableSegmentedDictionary.CreateBuilder<SingleNamespaceDeclaration, AliasesAndUsings>(ReferenceEqualityComparer.Instance);
+#endif
+
+                foreach (var singleDeclaration in _mergedDeclaration.Declarations)
+                {
+                    if (singleDeclaration.HasExternAliases || singleDeclaration.HasGlobalUsings || singleDeclaration.HasUsings)
+                    {
+                        builder.Add(singleDeclaration, new AliasesAndUsings());
+                    }
+#if DEBUG
+                    else
+                    {
+                        builderForAsserts.Add(singleDeclaration, new AliasesAndUsings());
+                    }
+#endif
+                }
+
+                RoslynImmutableInterlocked.InterlockedInitialize(ref _aliasesAndUsings_doNotAccessDirectly, builder.ToImmutable());
+#if DEBUG
+                RoslynImmutableInterlocked.InterlockedInitialize(ref _aliasesAndUsingsForAsserts_doNotAccessDirectly, builderForAsserts.ToImmutable());
+#endif
+            }
+
+            dictionary.TryGetValue(declaration, out var result);
+            Debug.Assert(result is not null);
+
+            return result ?? new AliasesAndUsings();
```

Comment:
this seems suspect.  the presumption i saw from all callers is that they only call when they expect an element to be in the dict.  so falling back to this fresh instance seems bad.

Reply:
Changed to throw in this case.